x-pack/plugins/fleet/server/types/models/agent_policy.ts

```diff
@@ -79,18 +90,34 @@ export const AgentPolicyBaseSchema = {
   ),
   is_protected: schema.maybe(schema.boolean()),
   overrides: schema.maybe(
-    schema.nullable(
+    schema.oneOf([
+      schema.literal(null),
       schema.recordOf(schema.string(), schema.any(), {
         validate: (val) => {
           if (Object.keys(val).some((key) => key.match(/^inputs(\.)?/))) {
             return 'inputs overrides is not allowed';
           }
         },
-      })
-    )
+        meta: {
+          description:
+            'Override settings that are defined in the agent policy. Input settings cannot be overridden. The override option should be used only in unusual circumstances and not as a routine procedure.',
+        },
+      }),
+    ])
   ),
   ...getSettingsAPISchema('AGENT_POLICY_ADVANCED_SETTINGS'),
-  supports_agentless: schema.maybe(schema.boolean({ defaultValue: false })),
+  supports_agentless: schema.maybe(
+    schema.oneOf([
+      schema.literal(null),
+      schema.boolean({
+        defaultValue: false,
+        meta: {
+          description:
+            'Indicates whether the agent policy supports agentless integrations. Only allowed in a serverless environment.',
```

Comment:
> Only allowed in a serverless environment.

Is this still true? I thought there was a plan to have agentless on cloud (stateful) as well

Reply:
updated

Reply:
Thank you!